src/main/java/com/yahoo/bullet/bql/query/TypeChecker.java

```diff
@@ -359,8 +359,12 @@
                 if (!errors.isEmpty()) {
                     return Optional.of(errors);
                 }
-                subType = rightType.getSubType();
-                if (subType != leftType && subType.getSubType() != leftType) {
+                if (Type.isComplexList(rightType) || Type.isComplexMap(rightType)) {
+                    subType = rightType.getSubType().getSubType();
+                } else {
+                    subType = rightType.getSubType();
+                }
```

Comment:
Would we just error out before this fix?

Reply:
Not a fix. I wanted to support matching numeric types, but previously the conditional checked if subType or subType.getSubType() matched the left side. Making that conditional fit numerics would've been needlessly complicated so I just changed the code to just initially get the primitive subtype.